### DISCUSSION TITLE
How do I download the cover image (or any image) with the fileId from the local API?

### DISCUSSION THREAD

**Discussion by l3d00m** [librespot-org/librespot-java#405] [Q&A] [answered]
I'm using librespot-java's API to get track metadata, which works pretty nicely. 

Is there an easy way to download the image? As far as I see it, the local API returns only the `fileId`, which is a HEX string. Can I somehow convert that to a URL on spotfiy's servers without needing to call the Spotify API again?

If I simply prepend `https://i.scdn.co/image/` to the URL it doesn't seem to work. I guess that's because I need to be authorized to access that URL?

**Comment by devgianlu** [comment, marked as answer]
Have a look at https://github.com/librespot-org/librespot-java/blob/dev/player/src/main/java/xyz/gianlu/librespot/player/Player.java#L787

That endpoint doesn't require authentication.

**Reply by l3d00m** [reply]
I tried that, but I missed a single step:

The `fileId` that is returned in the Image object has to be converted to lower case, it's in upper case by default! If any character is upper case, spotify doesn't return the image: i.e. http://i.scdn.co/image/AB67616D00001E02057004096A50CFA9BAEF86C0 does not work, but http://i.scdn.co/image/ab67616d00001e02057004096a50cfa9baef86c0 does.

Thanks for the link :)

Btw: I'm using it for https://github.com/l3d00m/MMM-NowPlayingOnLibrespot, so it's possible to display the current song on a Magic Mirror by using the librespot java API

**Reply by devgianlu** [reply]
Awesome!
